### PR TITLE
chore(vue): refactor code for ToggleButton

### PR DESCRIPTION
**Pull request by jyash97**
- Replace `this.$props.item` to `this.item`
- To access all props we may need to pass `this.$props`

### CODE REVIEW

File: packages/vue/src/components/list/ToggleButton.jsx [hunk 1]
```diff
@@ -34,9 +34,8 @@ const ToggleButton = {
 		renderItem: types.func,
 	},
 	data() {
-		const props = this.$props;
-		const value = this.selectedValue || props.value || props.defaultSelected || [];
-		const currentValue = ToggleButton.parseValue(value, props);
+		const value = this.selectedValue || this.value || this.defaultSelected || [];
+		const currentValue = ToggleButton.parseValue(value, this.$props);
```
Comment (bietkul) [inline packages/vue/src/components/list/ToggleButton.jsx:38]: Here you're passing `this.$props` in `parseValue`.

File: packages/vue/src/components/list/ToggleButton.jsx [hunk 2]
```diff
@@ -45,73 +44,72 @@ const ToggleButton = {
...
 		});
 	},
 	beforeDestroy() {
-		this.removeComponent(this.$props.componentId);
+		this.removeComponent(this.componentId);
 	},
 	watch: {
 		defaultSelected(newVal) {
-			this.setValue(ToggleButton.parseValue(newVal, this.$props));
+			this.setValue(ToggleButton.parseValue(newVal));
```
Comment (bietkul) [inline packages/vue/src/components/list/ToggleButton.jsx:67]: But `this.$props` is missing here, why is there inconsistency?

**Review by bietkul — CHANGES_REQUESTED** [review]
@jyash97  Can you do one thing, don't modify the `this.$props` passed in the methods because it might break the things which we'll be hard to debug in the future.
for example, don't change things like that `this.updateQuery(this.$data.currentValue, this.$props);` to just `this.updateQuery(this.currentValue);`, you removed the second param completely which made the method less customizable.
What if I want to set something forcefully in second param?
It is not possible now.
We're doing such things in other components.
Just change the way of accessing the properties for example if somewhere we're using `this.$data.currentValue` or `this.$props.currentValue` then change it to just `this.currentValue`.

**Comment by jyash97** [comment]
After rethinking again I guess we should avoid this in order to not break things and the current set of things are working well and we will need `this.$props` in order to pass all props value. So lets close this PR.

**Pull request closed without merging**